src/Microsoft.Extensions.Logging.Log4Net.AspNetCore/Log4NetProvider.cs

```diff
@@ -296,49 +300,57 @@ private void SetOptionsIfValid(Log4NetProviderOptions options)
         /// Configures the log4net library using the available configuration data.
         /// </summary>
         /// <param name="assembly">The assembly to be used on the configuration.</param>
-        private Log4NetProvider ConfigureLog4NetLibrary(Assembly assembly)
+#if NETCOREAPP1_1
+        private void ConfigureLog4NetLibrary(Assembly assembly)
```

Comment:
This approach is unnecessary from my point of view. Maybe you can implement it, if you want to sepparate the logic from Net Core 1.1 from the others, as an optional parameter that you check at the beginning of the method.

This is:
```
private void ConfigureLog4NetLibrary(Assembly assembly = null)
{
    bool useAssemblyInformation = assembly != null;

    // ...
    if (!options.ExternalConfigurationSetup)
    {
        string fileNamePath = useAssemblyInformation ? CreateLog4NetFilePath(assembly) : CreateLog4NetFilePath();
        /...
}
```

This comment applies also for the definition of the `CreateLog4NetFilePath` method.